packages/contracts/src/contracts/current/protocol/Exchange/libs/LibOrder.sol

```diff
@@ -73,17 +73,13 @@ contract LibOrder {
         view
         returns (bytes32 orderHash)
     {
-        // TODO: EIP712 is not finalized yet
-        // Source: https://github.com/ethereum/EIPs/pull/712
-        orderHash = keccak256(
-            DOMAIN_SEPARATOR_SCHEMA_HASH,
-            keccak256(address(this)),
-            ORDER_SCHEMA_HASH,
+        orderHash = createEIP712Message(
```

Comment:
The way EIP712 is intended is that we leave out `createEIP712Message` here and instead wrap the message in `SignatureValidator`.

Pro: getOrderHash will be simpler and cheaper.
Con: orderHash will not include the domain separator (but the signature still would).

Reply:
I've split the two so we can decide what to do without much modification